game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java

```diff
@@ -375,4 +399,50 @@ public void changeChanceDecrementOrIncrementOnSuccessOrFailure(final IDelegateBr
       delegateBridge.addChange(ChangeFactory.attachmentPropertyChange(this, newChance, CHANCE));
     }
   }
+
+  private static Map<String, Function<IAttachment, AttachmentProperty<?>>> getPopulatedAttachmentMap() {
+    return ImmutableMap.<String, Function<IAttachment, AttachmentProperty<?>>>builder()
+        .put("conditions",
+            ofCast(a -> AttachmentProperty.of(
```

Comment:
I'm not sure what exactly you are proposing 🤔

Reply:
Instead of `AttachmentProperty.of(` taking many parameters, we'd have something like:
```
ofCase(a -> AttachmentProperty.ofConditionsBuilder(a))
```
Which could be rewritten as a method reference.